docs/src/modules/java-protobuf/pages/workflows.adoc

```diff
@@ -0,0 +1,200 @@
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=timeouts]
+----
+<1> Sets a workflow global timeout.
+<2> Sets a default timeout for all workflow steps.
+
+A default step timeout can be overridden in step builder.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=step-timeout]
+----
+<1> Overrides the step timeout for a specific step.
+
+=== Recover strategy
+
+It's time to define what should happen in case of timeout or any other unhandled error.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=recover-strategy]
+----
+<1> Sets a failover transition in case of a workflow timeout.
+<2> Sets a default failover transition for all steps with maximum number of retries.
+<3> Overrides the step recovery strategy for the `deposit` step.
+
+
+NOTE: In case of a workflow timeout one last failover step can be performed. Transitions from that failover step will be ignored.
+
+=== Compensation
+
+The idea behind the Workflow error handling is that workflows should only fail due to unknown errors during execution. In general, you should always write your workflows so that they do not fail on any known edge cases. If you expect an error, it's better to be explicit about it, possibly with your domain types. Based on this information and the flexible Workflow API you can define a compensation for any workflow step.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/api/TransferWorkflow.java
+----
+include::example$java-protobuf-transfer-workflow-compensation/src/main/java/com/example/transfer/api/TransferWorkflow.java[tag=compensation]
+----
+<1> Explicit deposit call result type `DepositResult`.
+<2> Finishes workflow as completed, in the case of a successful deposit.
+<3> Launches compensation step to handle deposit failure. The `"withdraw"` step must be reversed. Compensation step is just any other step, with the same set of functionalities.
+<4> Correct compensation can finish the workflow.
+<5> Any other result might be handled by a default recovery strategy.
+
+Compensating a workflow step (steps) might involve multiple logical steps and thus is part of the overall business logic that must be defined within the workflow itself. For simplicity, in the example above, the compensation is applied only to `withdraw` step. Whereas `deposit` step itself might also require a compensation. In case of a step timeout we can't be certain about step successful or error outcome.
```

Comment:
"In case of a step timeout we can’t be certain about step successful or error outcome" Could you elaborate on that?

Reply:
For instance we are calling some service, in case of a timeout there are 2 options:
 - request failed
 - request was actually successful but take more time

Reply:
Right, right. 
I think is important the fact that it would happen when calling a service. Maybe some rephrasing? Here's one possibility.

```
When calling another service in a step, if the step times out, we can't be sure if the call was successful or not.
```